src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -0,0 +1,267 @@
+    private final JButton searchModeButton = new JButton();
+    private final JToggleButton caseSensitive;
+    private final JToggleButton regularExp;
+
+    private final SearchQueryHighlightObservable searchQueryHighlightObservable = new SearchQueryHighlightObservable();
+
+    private SearchWorker searchWorker;
+    private AutoCompleteSupport<String> autoCompleteSupport = new AutoCompleteSupport<>(searchField);
+
+    private final SearchPreferences searchPreferences;
+    private SearchDisplayMode searchDisplayMode;
+
+
+    public GlobalSearchBar() {
+        super();
+        searchPreferences = new SearchPreferences(Globals.prefs);
+        searchDisplayMode = searchPreferences.getSearchMode();
+
+        currentResults.setFont(currentResults.getFont().deriveFont(Font.BOLD));
+
+        if (OS.OS_X) {
+            searchField.putClientProperty("JTextField.variant", "search");
+        }
+        searchField.setColumns(30);
+
+        regularExp = new JToggleButton(IconTheme.JabRefIcon.REG_EX.getSmallIcon(),
+                searchPreferences.isRegularExpression());
+        regularExp.setToolTipText(Localization.lang("regular expression"));
+        regularExp.addActionListener(e -> {
+            searchPreferences.setRegularExpression(regularExp.isSelected());
+            performSearch();
+        });
+
+        caseSensitive = new JToggleButton(IconTheme.JabRefIcon.CASE_SENSITIVE.getSmallIcon(),
+                searchPreferences.isCaseSensitive());
+        caseSensitive.setToolTipText(Localization.lang("Case sensitive"));
+        caseSensitive.addActionListener(e -> {
+            searchPreferences.setCaseSensitive(caseSensitive.isSelected());
+            performSearch();
+        });
+
+        updateSearchModeButtonText();
+        searchModeButton.addActionListener(l -> {
+            int nextSearchMode = (searchDisplayMode.ordinal() + 1) % SearchDisplayMode.values().length;
+            searchDisplayMode = SearchDisplayMode.values()[nextSearchMode];
+            searchPreferences.setSearchMode(searchDisplayMode);
+            updateSearchModeButtonText();
+            performSearch();
+        });
+
+        clearSearchButton.setToolTipText(Localization.lang("Clear"));
+        clearSearchButton.addActionListener(l -> endSearch());
+
+        searchField.addFocusListener(Globals.focusListener);
+        searchField.addActionListener(e -> performSearch());
+        JTextFieldChangeListenerUtil.addChangeListener(searchField, e -> performSearch());
+        searchField.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.getExtendedKeyCode() == KeyEvent.VK_ESCAPE) {
+                    endSearch();
+                }
+            }
+        });
+
+        autoCompleteSupport.install();
+
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
+        toolBar.setFloatable(false);
+        toolBar.add(currentResults);
+        toolBar.add(searchIcon);
```

Comment:
I guess this only affects mac os x as it automatically adds a icon when you write
``
if (OS.OS_X) {
    searchField.putClientProperty("JTextField.variant", "search");
}
``